### PR TITLE
Register S3 actions in plugin.xml, move localization file

### PR DESCRIPTION
**Pull request by abrooksv**
* Move our localization file to match IJ sdk conventions to allow action localization to work
* Use shortcut of delete so we don't need to manage it ourselves

## Screenshots (if appropriate)
With selection:
Screen Shot 2021-02-26 at 11 42 46 AM

No selection:
Screen Shot 2021-02-26 at 11 55 48 AM

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/resources/META-INF/plugin.xml [hunk 1]
```diff
@@ -380,6 +379,40 @@ with what features/services are supported.
...
+        <group id="aws.toolkit.s3viewer.toolbar.pt2">
+            <action class="software.aws.toolkits.jetbrains.services.s3.objectActions.NewFolderAction"/>
+            <action class="software.aws.toolkits.jetbrains.services.s3.objectActions.RenameObjectAction" use-shortcut-of="RenameElement"/>
+            <action class="software.aws.toolkits.jetbrains.services.s3.objectActions.DeleteObjectAction" use-shortcut-of="$Delete"/>
+            <separator/>
+            <action class="software.aws.toolkits.jetbrains.services.s3.objectActions.RefreshTreeAction"/>
+        </group>
+
+        <group id="aws.toolkit.s3viewer.toolbar">
+            <reference ref="aws.toolkit.s3viewer.toolbar.pt1"/>
```
Comment (rli) [inline jetbrains-core/resources/META-INF/plugin.xml:400]: something more descriptive that `pt1`/`pt2`?

Comment (abrooksv) [reply jetbrains-core/resources/META-INF/plugin.xml:400]: I couldnt think of a name -.-

Comment (rli) [reply jetbrains-core/resources/META-INF/plugin.xml:400]: `aws.toolkit.s3viewer.object.transfer`
`aws.toolkit.s3viewer.object.management`

but then viewing objects isn't really related to transfer...

**Review by rli — APPROVED** [review]

File: jetbrains-core/resources/META-INF/plugin.xml (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/resources/META-INF/plugin.xml]: Broke it up into more groups, what you think?

**Review by rli — APPROVED** [review]

File: jetbrains-core/resources/META-INF/plugin.xml [hunk 2]
```diff
@@ -380,6 +379,51 @@ with what features/services are supported.
...
+        </group>
+
+        <group id="aws.toolkit.s3viewer.mutateActions">
+            <action class="software.aws.toolkits.jetbrains.services.s3.objectActions.NewFolderAction"/>
+            <action class="software.aws.toolkits.jetbrains.services.s3.objectActions.RenameObjectAction" use-shortcut-of="RenameElement"/>
+            <action class="software.aws.toolkits.jetbrains.services.s3.objectActions.DeleteObjectAction" use-shortcut-of="$Delete"/>
+        </group>
+
+        <group id="aws.toolkit.s3viewer.miscActions">
+            <action class="software.aws.toolkits.jetbrains.services.s3.objectActions.RefreshTreeAction"/>
```
Comment (hunterwerlla) [inline jetbrains-core/resources/META-INF/plugin.xml:399]: `use-shortcut-of="Refresh"`?

File: jetbrains-core/resources/META-INF/plugin.xml [hunk 3]
```diff
@@ -380,6 +379,51 @@ with what features/services are supported.
...
+            <action class="software.aws.toolkits.jetbrains.services.s3.objectActions.UploadObjectAction"/>
+        </group>
+
+        <group id="aws.toolkit.s3viewer.versionActions">
+            <action class="software.aws.toolkits.jetbrains.services.s3.objectActions.ViewObjectVersionAction"/>
+        </group>
+
+        <group id="aws.toolkit.s3viewer.mutateActions">
+            <action class="software.aws.toolkits.jetbrains.services.s3.objectActions.NewFolderAction"/>
+            <action class="software.aws.toolkits.jetbrains.services.s3.objectActions.RenameObjectAction" use-shortcut-of="RenameElement"/>
```
Comment (hunterwerlla) [inline jetbrains-core/resources/META-INF/plugin.xml:394]: Cool, we definitely need to use `use-shortcut-of` more

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DeleteObjectAction.kt [hunk 4]
```diff
@@ -29,35 +29,30 @@ class DeleteObjectAction : S3ObjectAction(message("s3.delete.object.action"), Al
...
-private fun deleteNodes(project: Project, treeTable: S3TreeTable, nodes: List<S3TreeObjectNode>) {
-    val response = Messages.showOkCancelDialog(
-        project,
-        message("s3.delete.object.description", nodes.size),
-        message("s3.delete.object.action"),
-        message("general.delete"),
-        message("s3.delete.object.cancel"),
-        Messages.getWarningIcon()
-    )
+    private fun deleteNodes(project: Project, treeTable: S3TreeTable, nodes: List<S3TreeObjectNode>) {
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DeleteObjectAction.kt:33]: This function is used in multiple places, moving it into the class breaks that which will break tests and runtime

File: jetbrains-core/resources/META-INF/plugin.xml (hunk 2 above)
Comment (abrooksv) [reply jetbrains-core/resources/META-INF/plugin.xml:399]: Ah we didnt have that before so didn't think to add it, good idea

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DeleteObjectAction.kt (hunk 4 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DeleteObjectAction.kt:33]: It shouldn't, its not used anywhere else now that we handle delete through the shortcut isntead of key listener

**Review by hunterwerlla — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DeleteObjectAction.kt (hunk 4 above)
Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DeleteObjectAction.kt:33]: Cool, I wasn't sure exactly where else it was used.

**Pull request merged**